### PR TITLE
2263 update ro spec

### PR DESCRIPTION
**Pull request by andi-huber**
I've not touched any of the site stuff yet. Also I've not executed 'sync adoc'.

- all XSD schemas bumped to v2/2.0
- BookmarkObjectState removed from schema
- which allowed to remove the Oid States (transient/viewmodel/persistent)
- maven build seems to work
- demo-app seems to work
- 6 java.time temporals have been added to the schema

---
This change is [Reviewable](https://reviewable.io/reviews/apache/isis/171)

### CODE REVIEW

File: api/schema/src/main/resources/org/apache/isis/schema/chg/chg-2.0.xsd [hunk 1]
```diff
@@ -24,11 +24,11 @@
            xmlns:com="http://isis.apache.org/schema/common"
 >
 
-    <xs:import namespace="http://isis.apache.org/schema/common" schemaLocation="../common/common-1.1.xsd"/>
+    <xs:import namespace="http://isis.apache.org/schema/common" schemaLocation="../common/common-2.0.xsd"/>
```
Comment (danhaywood) [inline api/schema/src/main/resources/org/apache/isis/schema/chg/chg-2.0.xsd]: the "default" attribute of these two elements:

<xs:element name="majorVersion" type="xs:string" minOccurs="0" maxOccurs="1" default="1">
<xs:element name="minorVersion" type="xs:string" minOccurs="0" maxOccurs="1" default="1">

should be changed in line version (ie to default="2" and default="0." respectively).

This applies to all of the updated .xsd files.

File: core/metamodel/src/main/java/org/apache/isis/core/metamodel/adapter/oid/Oid.java [hunk 2]
```diff
@@ -41,24 +41,16 @@
...
-
-    boolean isViewModel();
-
-    boolean isPersistent();
-
     default boolean isValue() {
         return false; // default, only overridden by Oid_Value
     }
     
+    @Deprecated
```
Comment (danhaywood) [inline core/metamodel/src/main/java/org/apache/isis/core/metamodel/adapter/oid/Oid.java]: I presume there was a reason you kept this in?

File: core/metamodel/src/main/java/org/apache/isis/core/metamodel/adapter/oid/Oid_Marshaller.java [hunk 3]
```diff
@@ -88,14 +87,10 @@
...
 
-    // -- public constants
-    public static final String VIEWMODEL_INDICATOR =
-            Bookmark.ObjectState.VIEW_MODEL.getCode(); // "*"
-
-
-    // -- private constants
-    private static final String TRANSIENT_INDICATOR =
-            Bookmark.ObjectState.TRANSIENT.getCode() ; // "!"
+    @Deprecated
```
Comment (danhaywood) [inline core/metamodel/src/main/java/org/apache/isis/core/metamodel/adapter/oid/Oid_Marshaller.java]: I wonder why these are required?  The objectType will allow us to determine if this is a view model or an entity, so why this flag also?

File: core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/datejdk8local/Jdk8LocalDateUtil.java [hunk 4]
```diff
@@ -1,170 +0,0 @@
-/*
```
Comment (danhaywood) [inline core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/datejdk8local/Jdk8LocalDateUtil.java]: What's happened to this file?  It doesn't seem to have moved, is just deleted,.   Ditto the other classes deleted in the same area.

File: core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/datejdk8local/Jdk8LocalDateValueFacetUsingSemanticsProviderFactory.java [hunk 5]
```diff
@@ -1,45 +0,0 @@
-/*
```
Comment (danhaywood) [inline core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/datejdk8local/Jdk8LocalDateValueFacetUsingSemanticsProviderFactory.java]: why deleted?

File: core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/datejdk8local/Jdk8LocalDateValueSemanticsProvider.java [hunk 6]
```diff
@@ -1,252 +0,0 @@
-/*
```
Comment (danhaywood) [inline core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/datejdk8local/Jdk8LocalDateValueSemanticsProvider.java]: why deleted?

File: core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/datejdk8local/TimeFormatter.java [hunk 7]
```diff
@@ -1,27 +0,0 @@
-/*
```
Comment (danhaywood) [inline core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/datejdk8local/TimeFormatter.java]: why deleted?

File: core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/temporal/TemporalAdjust.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
+package org.apache.isis.core.metamodel.facets.value.temporal;
```
Comment (danhaywood) [inline core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/temporal/TemporalAdjust.java]: unit tests?

File: core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/temporal/TemporalAdjustUtil.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+import java.time.temporal.Temporal;
+import java.util.StringTokenizer;
+import java.util.function.BiFunction;
+
+import lombok.val;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+final class TemporalAdjustUtil {
```
Comment (danhaywood) [inline core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/temporal/TemporalAdjustUtil.java]: couldn't these static methods be on TempralAdjust itself?

File: core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/temporal/TemporalValueFacet.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+ * @since 2.0
+ *
+ * @param <T> implementing {@link java.time.temporal.Temporal} type
+ */
+public interface TemporalValueFacet<T extends Temporal> extends Facet {
+
+    T temporalValue(ManagedObject object);
+    ManagedObject createValue(T temporal);
+    
+//    TemporalCategory getTemporalCategory();
```
Comment (danhaywood) [inline core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/temporal/TemporalValueFacet.java]: remove these commented out lines?

File: api/applib/src/main/java/org/apache/isis/applib/util/JaxbAdapters.java [hunk 11]
```diff
@@ -156,6 +161,20 @@ public String marshal(java.sql.Timestamp v) throws Exception {
 
     }
 
+    public static final class LocalTimeAdapter extends XmlAdapter<String, LocalTime> {
```
[the remark this answers is not in the document]
Comment (andi-huber) [reply api/applib/src/main/java/org/apache/isis/applib/util/JaxbAdapters.java]: good point; I'm adding these; I just discovered all these adapters are not null-safe; also fixing this

File: core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/temporal/TemporalAdjust.java (hunk 8 above)
Comment (andi-huber) [reply core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/value/temporal/TemporalAdjust.java]: TemporalAdjustTest added

**Pull request merged**